kong/pdk/private/node.lua

```diff
@@ -0,0 +1,88 @@
+local utils = require "kong.tools.utils"
+local pl_file = require "pl.file"
+local pl_path = require "pl.path"
+local pl_dir = require "pl.dir"
+
+local ngx = ngx
+local fmt = string.format
+
+local cached_node_id
+
+local function node_id_filename(prefix)
+  return pl_path.join(prefix, "kong.id")
+end
+
+
+local function initialize_node_id(prefix)
+  if not pl_path.exists(prefix) then
+    local ok, err = pl_dir.makepath(prefix)
+    if not ok then
+      return nil, fmt("failed to create directory %s: %s", prefix, err)
+    end
+  end
+
+  local filename = node_id_filename(prefix)
+
+  if not pl_path.exists(filename) then
+    local id = utils.uuid()
+    ngx.log(ngx.DEBUG, "persisting node_id (", id, ") to ", filename)
+
+    local ok, write_err = pl_file.write(filename, id)
+    if not ok then
+      return nil, fmt("failed to persist node_id to %s: %s", filename, write_err)
+    end
+    cached_node_id = id
```

Comment:
@vm-001 After occasionally checking this part of code, I have a question: the node_id gets cached only when the file doesn't exist, right? Can we cache it even if the file already exists? If so, `load_node_id` will benefit from this.